qiskit/transpiler/passes/layout/layout_2q_distance.py

```diff
@@ -52,12 +52,18 @@ def run(self, dag):
         if layout is None:
             return
 
+        if self.coupling_map is None or len(self.coupling_map.graph) == 0:
+            self.property_set[self.property_name] = 0
+            return
+
+        self.coupling_map.compute_distance_matrix()
+
         sum_distance = 0
 
         for gate in dag.two_qubit_ops():
-            physical_q0 = layout[gate.qargs[0]]
-            physical_q1 = layout[gate.qargs[1]]
+            physical_q0 = layout._v2p[gate.qargs[0]]
+            physical_q1 = layout._v2p[gate.qargs[1]]
```

Comment:
Can we fetch `layout.get_virtual_bits()` once outside the loop?
```python
        virtual_physical_map = layout.get_virtual_bits()
        for gate in dag.two_qubit_ops():
            physical_q0 = virtual_physical_map[gate.qargs[0]]
            physical_q1 = virtual_physical_map[gate.qargs[1]]
```

Reply:
Also, would it make sense to deprecate the `Layout.__getitem__` if it's known to be a hidden performance pitfall?

Reply:
Yeah, that should work. Done in: https://github.com/Qiskit/qiskit-terra/pull/7036/commits/0154230fa4c260d7942b11f39d1ca07456fa612e

Reply:
As for the deprecation I think that probably makes sense, but we maybe should do that in a follow up because I don't know how widely used it is

Reply:
Opened https://github.com/Qiskit/qiskit-terra/issues/7290 to track.